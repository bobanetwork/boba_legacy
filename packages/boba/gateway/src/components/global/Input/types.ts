import { ReactNode } from 'react'

export interface InputProps {
  placeholder: string
  type: string
  onChange: (e: React.ChangeEvent<HTMLInputElement>) => void
  value: any
  label?: ReactNode
}
