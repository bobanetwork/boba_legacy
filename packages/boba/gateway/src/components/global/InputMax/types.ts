export interface MaxInputTypes {
  max: number
  initialValue?: number
  onValueChange: (value: number) => void
}
