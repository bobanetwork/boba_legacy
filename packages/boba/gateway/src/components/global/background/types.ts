export enum BackgroundPosition {
  TOP = 'top',
  CENTER = 'center',
}

export interface BackgroundProps {
  position: BackgroundPosition
}

export const RoutesWithBackgroundPositionAtTop = ['/bridge', '/']
