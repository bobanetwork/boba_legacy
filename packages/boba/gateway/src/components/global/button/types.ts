import { ReactNode } from 'react'

type ButtonType = {
  disable?: boolean
  loading?: boolean
  small?: boolean
  outline?: boolean
  tiny?: boolean
  transparent?: boolean
}

export interface ButtonTypes extends ButtonType {
  label: ReactNode | string
  style?: Record<string, string>
  className?: string
  onClick?: () => void
}
