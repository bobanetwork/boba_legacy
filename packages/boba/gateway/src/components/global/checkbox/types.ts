export interface CheckboxWithLabelProps {
  label?: string
  checked?: boolean
  onChange: (isChecked: boolean) => void
}
