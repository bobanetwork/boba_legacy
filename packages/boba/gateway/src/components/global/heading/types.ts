import { ReactNode } from 'react'

export type VariantType = 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6'

export interface HeadingProps {
  variant: VariantType
  color?: string
  children?: ReactNode
  className?: string
}

export interface HeadingStyleProps {
  size: string
  lineHeight: string
  fontWeight: number
}
