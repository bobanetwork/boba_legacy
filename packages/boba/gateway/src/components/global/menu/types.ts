import { ComponentType, ReactElement, ReactNode } from 'react'

export interface OptionProps {
  label?: string
  component?: ReactElement<any>
  onClick?: (d: any) => void
}

export interface MenuProps {
  name?: string
  label?: ReactNode | string
  children?: ReactNode
  options: Array<OptionProps>
}
