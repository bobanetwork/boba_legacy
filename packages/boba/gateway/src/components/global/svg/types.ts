export interface SvgTypes {
  src: string
  fill: string
}
