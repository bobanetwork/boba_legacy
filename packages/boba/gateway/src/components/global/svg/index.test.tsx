import React from 'react'
import { render } from '@testing-library/react'
import { SvgTypes } from './types'
import CustomThemeProvider from 'themes'
import { Svg } from './'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import styled from 'styled-components'
const mockStore = configureStore()
const ArrowOnHover = styled.div`
  svg {
    &:hover {
      fill: red;
    }
  }
`

const renderSvg = (props: SvgTypes) => {
  return render(
    <Provider
      store={mockStore({
        ui: {
          theme: 'light',
        },
      })}
    >
      <CustomThemeProvider>
        <Svg {...props} />
      </CustomThemeProvider>
    </Provider>
  )
}

const renderWrappedSvg = (props: SvgTypes) => {
  return render(
    <Provider
      store={mockStore({
        ui: {
          theme: 'light',
        },
      })}
    >
      <CustomThemeProvider>
        <ArrowOnHover>
          <Svg {...props} />
        </ArrowOnHover>
      </CustomThemeProvider>
    </Provider>
  )
}

describe('Svg', () => {
  test('Render Svg File', () => {
    const { asFragment } = renderSvg({
      fill: '#ccc',
      src: 'static/media/src/images/icons/arrowdown.svg',
    })
    expect(asFragment()).toMatchSnapshot()
  })
  test('Render Wrapped Svg file', () => {
    const { asFragment } = renderWrappedSvg({
      fill: 'yellow',
      src: 'static/media/src/images/icons/arrowdown.svg',
    })
    expect(asFragment()).toMatchSnapshot()
  })
})
