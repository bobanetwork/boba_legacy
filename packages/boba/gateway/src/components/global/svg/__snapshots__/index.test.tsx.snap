// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Svg Render Svg File 1`] = `
<DocumentFragment>
  <div>
    <div>
      <div
        data-src="static/media/src/images/icons/arrowdown.svg"
        src=""
      />
    </div>
  </div>
</DocumentFragment>
`;
